### PR TITLE
[PWN-6235] Update action button text due to state and make viewModel …

### PR DESCRIPTION
**Pull request by lisemyon**
…as MainActor

## Description of the changes
- https://p2pvalidator.atlassian.net/browse/PWN-6235

### CODE REVIEW

File: p2p_wallet/Scenes/Main/Send/Input/SendInputViewModel.swift [hunk 1]
```diff
@@ -199,9 +199,9 @@ private extension SendInputViewModel {
         $feeWallet
             .sinkAsync { [weak self] newFeeToken in
                 guard let self = self else { return }
-                await MainActor.run { self.isFeeLoading = true }
+                self.isFeeLoading = true
                 _ = await self.stateMachine.accept(action: .changeFeeToken(newFeeToken.token))
```
[the remark this answers is not in the document]
Comment (lisemyon) [reply p2p_wallet/Scenes/Main/Send/Input/SendInputViewModel.swift:203]: Yes it should have been but I have change the whole viewModel to MainActor. BaseViewModel gives this opportunity so it is in Main now

[the remark this answers is not in the document]
Comment (sidorov-panda) [reply p2p_wallet/Scenes/Main/Send/Input/SendInputViewModel.swift:203]: Got it, is it ok to put `await` on main then?

[the remark this answers is not in the document]
Comment (lisemyon) [reply p2p_wallet/Scenes/Main/Send/Input/SendInputViewModel.swift:203]: do you mean "await MainActor.run { self.isFeeLoading = true }" here? or in the new line?

[the remark this answers is not in the document]
Comment (lisemyon) [reply p2p_wallet/Scenes/Main/Send/Input/SendInputViewModel.swift:203]: stateMachine is another actor so it requires `await`

[the remark this answers is not in the document]
Comment (sidorov-panda) [reply p2p_wallet/Scenes/Main/Send/Input/SendInputViewModel.swift:203]: If the whole construction is Main is it safe to put `await` on Main. The logic behind await can change and the whole Main will wait for it. Maybe it's safer to use Task

**Pull request merged**